images/macos/provision/core/Configure-Toolset.ps1

```diff
@@ -0,0 +1,60 @@
+################################################################################
+##  File:  Configure-Toolset.ps1
+##  Team:  CI-Build
+##  Desc:  Configure toolset
+################################################################################
+
+Import-Module "~/image-generation/helpers/Common.Helpers.psm1"
+
+function Get-ToolsetToolFullPath
+{
+    param
+    (
+        [Parameter(Mandatory)] [string] $ToolName,
+        [Parameter(Mandatory)] [string] $ToolVersion,
+        [Parameter(Mandatory)] [string] $ToolArchitecture
+    )
+
+    $toolPath = Join-Path -Path $env:AGENT_TOOLSDIRECTORY -ChildPath $toolName
+    $toolPathVersion = Join-Path -Path $toolPath -ChildPath $toolVersion
+    $foundVersion = Get-Item $toolPathVersion | Sort-Object -Property {[version]$_.name} -Descending | Select-Object -First 1
+    $installationDir = Join-Path -Path $foundVersion -ChildPath $toolArchitecture
+    return $installationDir
+}
+
+function Add-EnvironmentVariable
+{
+    param
+    (
+        [Parameter(Mandatory)] [string] $Name,
+        [Parameter(Mandatory)] [string] $Value,
+        [string] $FilePath = "${env:HOME}/.bashrc"
+    )
+
+    $envVar = "{0}={1}" -f $name, $value
+    Tee-Object -InputObject $envVar -FilePath $filePath -Append
+}
+
+$toolset = Get-Content -Path "$env:HOME/image-generation/toolset.json" -Raw | ConvertFrom-Json
```

Comment:
The script [toolset.ps1](https://github.com/actions/virtual-environments/blob/main/images/macos/provision/core/toolset.ps1#L9) contains the `Get-toolcachefromtoolset` function that does the same. Could you please also move it to the helpers and reuse it here?

Reply:
I got rid of  the `Get-ToolcacheFromToolset` function, I used `Get-ToolsetValue` instead